pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -712,14 +721,11 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundListenerForPortOrUDS(no
 		}
 
 		hasTLSContext = hasTLSContext || c.TLSContext != nil
-	}
-
-	// only apply inbound tls options when no authentication applies
-	if !hasTLSContext && listenerOpts.userTLSOpts != nil {
-		downstreamTLSContext := buildSidecarListenerTLSContext(listenerOpts.userTLSOpts, node.Metadata, listenerOpts.push.Mesh.SdsUdsPath)
-		for i := range allChains {
-			// set downstream tls context according to inbound tls options
-			allChains[i].TLSContext = downstreamTLSContext
+		// only apply inbound tls options when authn is disabled or permissive
+		// i.e. if there is no istio tls context, add one provided by user
+		if c.TLSContext == nil {
```

Comment:
This means for permissive mode, it is tls enforced

Reply:
Yes.. user provided TLS